src/ray/raylet/node_manager.cc

```diff
@@ -210,14 +210,46 @@ ray::Status NodeManager::RegisterGcs() {
   return ray::Status::OK();
 }
 
+void NodeManager::KillWorker(std::shared_ptr<Worker> worker) {
```

Comment:
it may make more sense to put this function in `worker_pool`, which is responsible for creating and killing workers.
And we'll also need to remove the worker from the pool.

Reply:
Yes that makes sense. The problem is that 'KillWorker()' requires io_service_ and RayConfig which the worker pool is not aware of.